langchain/src/chat_models/openai.ts

```diff
@@ -258,68 +255,94 @@ export class ChatOpenAI extends BaseChatModel implements OpenAIInput {
     const params = this.invocationParams();
     params.stop = stop ?? params.stop;
 
-    const { data } = await this.completionWithRetry({
-      ...params,
-      messages: messages.map((message) => ({
-        role: messageTypeToOpenAIRole(message._getType()),
-        content: message.text,
-      })),
-    });
-
-    if (params.stream) {
-      let role: ChatCompletionResponseMessageRoleEnum = "assistant";
-      const completion = await new Promise<string>((resolve, reject) => {
-        let innerCompletion = "";
-        const parser = createParser((event) => {
-          if (event.type === "event") {
-            if (event.data === "[DONE]") {
-              resolve(innerCompletion);
-            } else {
-              const response = JSON.parse(event.data) as {
-                id: string;
-                object: string;
-                created: number;
-                model: string;
-                choices: Array<{
-                  index: number;
-                  finish_reason: string | null;
-                  delta: {
-                    content?: string;
-                    role?: ChatCompletionResponseMessageRoleEnum;
+    const data = params.stream
+      ? await new Promise<CreateChatCompletionResponse>((resolve, reject) => {
+          let response: CreateChatCompletionResponse;
+          let rejected = false;
+          this.completionWithRetry(
+            {
+              ...params,
+              messages: messages.map((message) => ({
+                role: messageTypeToOpenAIRole(message._getType()),
+                content: message.text,
+              })),
+            },
+            {
+              responseType: "stream",
+              onmessage: (event) => {
+                if (event.data === "[DONE]") {
```

Comment:
could there ever be extra newlines or carriage returns here?

Reply:
yea I guess it doesn't hurt to call .trim, will add